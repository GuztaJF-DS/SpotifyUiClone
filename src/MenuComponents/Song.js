import { useParams } from "react-router-dom";
import JSONData from '../Data.json';
import SecondsToHours from '../MenuFunctions/SecondsToHours'
import { useDispatch } from "react-redux";
import CreateTrackList from "../MenuFunctions/CreateTrackList";

export default function Song(){
    let params=useParams()
    let Tracks=JSONData[params.id].TrackList;
    let AlbumInSeconds=0;

    let dispatch=useDispatch();

    for (var i=0;i<Tracks.length;i++){
        let DurationSplited=Tracks[i].Duration.split(':');
        let seconds=(parseInt(DurationSplited[0])*60)+parseInt(DurationSplited[1]);
        AlbumInSeconds+=seconds; 
    }

    let AlbumLength=SecondsToHours(AlbumInSeconds)

    return (
        <div>
            <div style={{display:'flex',marginBottom:'25px'}}>
                <img src={JSONData[params.id].Cover} alt={JSONData[params.id].Name} className="AlbumCoverBig"/>
                <div style={{display:'block',marginLeft:'15px'}}>
                    <h1>{JSONData[params.id].Name}</h1>
                    <h2>{JSONData[params.id].Artist}</h2>
                    <h3>{AlbumLength}</h3>
                </div>
            </div>
            
            <CreateTrackList JSONData={JSONData} AlbumId={params.id} dispatch={dispatch}/>
        </div>
    )
}