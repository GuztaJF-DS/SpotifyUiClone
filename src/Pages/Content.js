import React,{useState,useEffect} from 'react';
import Data from '../Data.json'
import {Link} from 'react-router-dom'

function Content(){
    const [carousel,setCarousel]=useState(0);
    const [TriggerCarousel,setTriggerCarousel]=useState("");

    useEffect(()=>{
        try{
            if(TriggerCarousel==="-"){
                if(carousel!==0){
                    setCarousel(carousel-1);
                }
            }
            if(TriggerCarousel==="+"){
                if((carousel+5)!==(parseInt(Data.length))){
                    setCarousel(carousel+1);
                }
            }
            setTriggerCarousel("")
        }catch(err){
            console.log(err);
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    },[TriggerCarousel])
    
    function MultipleJsx(){
        let rows = []
        for (let i = carousel; i < carousel+5; i++) {
        rows.push(
                <nav className='MusicGroup' key={i}>
                    <Link style={{textDecoration:"none", color:'#ecf2f9'}} to={`/Album/${i}`}>
                        <div className="MusicSquare">
                            <img src={Data[i].Cover} alt={Data[i].Name} className="AlbumCover"/>
                            <div>{Data[i].Name}</div>
                        </div>
                    </Link>
                </nav>
                )
        }
        return rows
    }

    return(
        <div style={{flexDirection:"row",marginBottom:"7px"}}>
            <div style={{fontSize:"25px"}}>
                Fileira Aleat√≥ria
            </div>

        <div className="ContentMusicRow">
            <button className='CarouselButton' onClick={()=>{setTriggerCarousel("-")}}>{'<'}</button>
                <MultipleJsx/>
            <button className='CarouselButton' onClick={()=>{setTriggerCarousel("+")}}>{'>'}</button>
        </div>
     </div>
    )
}

export default Content;