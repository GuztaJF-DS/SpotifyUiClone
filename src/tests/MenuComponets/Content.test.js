import React from "react";
import renderer from 'react-test-renderer';
import { Routes,Route } from 'react-router-dom';
import {BrowserRouter} from 'react-router-dom'

import Content from "../../MenuComponents/Content";
import Song from '../../MenuComponents/Song'

it('renders?',()=>{
    const tree=renderer.create(
        <BrowserRouter>
        <Routes>
        <Route path="/Album/:id" element={<Song />}>
        </Route>
        <Route path="/" element={<><Content/></>} />
      </Routes></BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})