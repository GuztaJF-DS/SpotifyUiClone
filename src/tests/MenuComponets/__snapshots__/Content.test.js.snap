// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders? 1`] = `
<div
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": "7px",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": "25px",
      }
    }
  >
    Fileira Aleat√≥ria
  </div>
  <div
    className="ContentMusicRow"
  >
    <button
      className="CarouselButton"
      onClick={[Function]}
    >
      &lt;
    </button>
    <nav
      className="MusicGroup"
    >
      <a
        href="/Album/0"
        onClick={[Function]}
        style={
          Object {
            "color": "#ecf2f9",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="MusicSquare"
        >
          <img
            alt="Magical Mystery Tour"
            className="AlbumCover"
            src="https://m.media-amazon.com/images/I/91LrXdPTwZL._AC_SL1500_.jpg"
          />
          <div>
            Magical Mystery Tour
          </div>
        </div>
      </a>
    </nav>
    <nav
      className="MusicGroup"
    >
      <a
        href="/Album/1"
        onClick={[Function]}
        style={
          Object {
            "color": "#ecf2f9",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="MusicSquare"
        >
          <img
            alt="Wish You Were Here"
            className="AlbumCover"
            src="https://m.media-amazon.com/images/I/810CGm6+GbL._AC_SL1500_.jpg"
          />
          <div>
            Wish You Were Here
          </div>
        </div>
      </a>
    </nav>
    <nav
      className="MusicGroup"
    >
      <a
        href="/Album/2"
        onClick={[Function]}
        style={
          Object {
            "color": "#ecf2f9",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="MusicSquare"
        >
          <img
            alt="Meddle"
            className="AlbumCover"
            src="https://theentreacordes.files.wordpress.com/2021/07/meddle.jpg"
          />
          <div>
            Meddle
          </div>
        </div>
      </a>
    </nav>
    <nav
      className="MusicGroup"
    >
      <a
        href="/Album/3"
        onClick={[Function]}
        style={
          Object {
            "color": "#ecf2f9",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="MusicSquare"
        >
          <img
            alt="Californication"
            className="AlbumCover"
            src="https://m.media-amazon.com/images/I/81kKCMLtpBL._AC_SL1200_.jpg"
          />
          <div>
            Californication
          </div>
        </div>
      </a>
    </nav>
    <nav
      className="MusicGroup"
    >
      <a
        href="/Album/4"
        onClick={[Function]}
        style={
          Object {
            "color": "#ecf2f9",
            "textDecoration": "none",
          }
        }
      >
        <div
          className="MusicSquare"
        >
          <img
            alt="Pinkerton"
            className="AlbumCover"
            src="https://www.udiscovermusic.com/wp-content/uploads/2019/09/Weezer-Pinkerton-album-cover-820.jpg"
          />
          <div>
            Pinkerton
          </div>
        </div>
      </a>
    </nav>
    <button
      className="CarouselButton"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
</div>
`;
